code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,177 @@
+     *
+     * @param map The map containing the data for [AepButton].
+     * @return An instance of [AepButton].
+     */
+    private fun createAepButton(map: Map<String, Any>): AepButton {
+        return AepButton(
+            id = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.INTERACT_ID, null),
+            actionUrl = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.ACTION_URL, null),
+            text = DataReader.optTypedMap(Any::class.java, map, MessagingConstants.ContentCard.UIKeys.TEXT, null).let { createAepText(it) }
+        )
+    }
+
+    /**
+     * Creates an instance of [AepDismissButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepDismissButton].
+     * @return An instance of [AepDismissButton].
+     */
+    private fun createAepDismissButton(map: Map<String, Any>): AepDismissButton {
+        return AepDismissButton(
+            style = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.STYLE, MessagingConstants.ContentCard.UIKeys.NONE)
+        )
+    }
+
+    /**
+     * Parses the given [ContentCardSchemaData] to create an [AepUITemplate].
+     *
+     * @param contentCardSchemaData The content card schema data containing the content map.
+     * @return The parsed [AepUITemplate] or null if parsing fails.
+     */
+    internal fun getTemplateModel(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+        try {
+            val contentMap =
+                contentCardSchemaData.content as? Map<String, Any> ?: throw IllegalArgumentException("Content map is null")
+            val id = DataReader.getString(contentMap, MessagingConstants.ContentCard.UIKeys.ID)
+
+            val title = DataReader.getTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.TITLE)?.let {
+                createAepText(it)
+            }
+            if (title == null || title.content.isEmpty()) {
+                return null
+            }
+            val body = DataReader.optTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.BODY, emptyMap())?.let {
+                createAepText(it)
+            }
+            val image = DataReader.optTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.IMAGE, emptyMap())?.let {
+                createAepImage(it)
+            }
+            val actionUrl = DataReader.optString(contentMap, MessagingConstants.ContentCard.UIKeys.ACTION_URL, null)
+            val buttons = DataReader.optTypedListOfMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.BUTTONS, emptyList()).map {
+                createAepButton(it)
+            }
+            val dismissBtn = DataReader.optTypedMap(Any::class.java, contentMap, MessagingConstants.ContentCard.UIKeys.DISMISS_BTN, emptyMap())?.let {
+                createAepDismissButton(it)
+            }
+            return SmallImageTemplate(
```

Comment:
The `template` type is inside the meta data field. You should be using that to decide which type of template to create. Please refer to this spec for more details
https://wiki.corp.adobe.com/display/adms/Small+Image+Template